liesel/goose/builder.py

```diff
@@ -129,8 +130,16 @@ class EngineBuilder:
     from your posterior distribution.
     """
 
-    def __init__(self, seed: int, num_chains: int):
-        keys = jax.random.split(jax.random.PRNGKey(seed), 3)
+    def __init__(self, seed: int | jax.Array, num_chains: int):
```

Comment:
the type annotation for a key should be `jax.random.KeyArray`. currently, that is just a type alias to array but it might change in the future.

Reply:
I used this at first however I got a `DeprecationWarning`, that's why I added the two issues above that deal with the switch away from using `jax.random.KeyArray` or `jax.random.PRNGKeyArray`. Here again some infos from the [docs](https://jax.readthedocs.io/en/latest/jep/9263-typed-keys.html#type-annotations-for-prng-keys).

Reply:
I missed that jax has implemented a new key style. I thought they just added more PRNG generators. To be consistent with the other code in Goose, i think it would be best to use `from liesel.goose.typing import KeyArray` which is just an alias to `typing.Any` and then do the test you proposed.

@jobrachem, at some point in the future, we need to change that everywhere in liesel. Also, we should use jax new jax new type Annotation. In Goose, we apparently use a type alias to Any as Array.

https://github.com/liesel-devs/liesel/blob/1ee02de035199d52448eb8788b13a720db5aafb6/liesel/goose/types.py#L22C8-L28